phone_numbers/phone.go

```diff
@@ -0,0 +1,97 @@
+package phonenumbers
+
+import (
+	"errors"
+	"strings"
+)
+
+// GetPhoneDetails returns the OperatorDetails for the given phoneNumber
+func GetPhoneDetails(phoneNumber string) (*OperatorDetails, error) {
+	if ok := IsPhoneValid(phoneNumber); !ok {
+		return nil, ErrInvalidFormat
+	}
+	prefix, err := GetOperatorPrefix(phoneNumber)
+	if err != nil {
+		return nil, err
+	}
+	return GetPrefixDetails(prefix)
+}
+
+// GetPrefixDetails returns the OperatorDetails of the given prefix
+func GetPrefixDetails(prefix string) (*OperatorDetails, error) {
+	for k, v := range MCIMap {
+		if k == prefix {
+			return &v, nil
+		}
```

Comment:
This part is repeating, there is an opportunity to make it cleaner and DRY

Reply:
I used a map to store all operators and range over them to make it cleaner

Reply:
Exactly what I meant, cool